lib/howl/interactions/line_selection.moon

```diff
@@ -0,0 +1,97 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import interact from howl
+import highlight from howl.ui
+import Matcher from howl.util
+
+line_match_highlighter = (editor) ->
+  local buffer
+  (selection, text, items) ->
+    if buffer
+      highlight.remove_all 'search', buffer
+      highlight.remove_all 'search_secondary', buffer
+
+    return unless selection
+
+    buffer = editor.buffer
+    line = buffer.lines[selection.line_nr]
+    local positions
+    if text and not text.is_empty
+      positions = Matcher.explain text, line.text
+
+    if positions
+      start_pos = line.start_pos
+      column = positions[1]
+      for hl_pos in *positions
+        highlight.apply 'search', buffer, start_pos + hl_pos - 1, 1
+    else
+      highlight.apply 'search', buffer, line.start_pos, line.end_pos - line.start_pos
+
+    -- highlight nearby secondary matches
+    if #items > 0 and not text.is_empty
+      local idx
+
+      for i, item in ipairs items
+        continue unless item.buffer == buffer
+        if item.line_nr >= editor.line_at_top - 5
+          idx = i
+          break
+
+      if idx
+        for i = idx, #items
+          continue unless items[i].buffer == buffer
+          nr = items[i].line_nr
+          if nr > editor.line_at_bottom + 5
+            break
+          if nr == selection.line_nr
+            continue
+          line = buffer.lines[nr]
+          start_pos = line.start_pos
+          positions = Matcher.explain text, line.text
+          if positions
+            for hl_pos in *positions
+              highlight.apply 'search_secondary', buffer, start_pos + hl_pos - 1, 1
+
+interact.register
+  name: 'select_line'
+  description: 'Selection for buffer lines'
+  handler: (opts) ->
+    lines = opts.lines
+    opts.lines = nil
```

Comment:
There's no harm in leaving this in but I didn't want to pass down unused, unnecessary options to select_location. I also didn't want anyone to use opts.lines within select_location, just because it happens to be available.